src/posix.mak

```diff
@@ -22,9 +22,15 @@ INSTALL_DIR=../../install
 SYSCONFDIR=/etc
 PGO_DIR=$(abspath pgo)
 
-C=backend
-TK=tk
-ROOT=root
+D = ddmd
+
+C=$D/backend
+TK=$D/tk
+ROOT=$D/root
+
+GENERATED = ../generated
+G = $(GENERATED)/$(OS)$(MODEL)
+$(shell mkdir -p $G)
```

Comment:
This is interesting. A few explanations follow.

1. Currently the code does not make the situation any worse than before - it just moves all generated files elsewhere, indiscriminately.

2. The structure `generated/$(OS)$(MODEL) ` (as opposed to Phobos' `generated/$(OS)/$(MODEL)`) was chosen by @RazvanN7 and I after discussing for a while and figuring there's no real need for two separate levels in the dir hierarchy.

3. We could actually simply use `generated/` without the OS or the model, BUT:
  * It is reasonable to generate 32 and 64 binaries on a given machine
  * Even on a given OS, there may be multiple "OS"s - consider generating Windows and Cygwin binaries on Windows. Are there any others? Not sure how important this is, but putting the OS in the dirname helps such situations.

4. Now (this is getting to @wilzbach's point), we have the situation in which changing CFLAGS and DFLAGS affects the generated binaries in many ways, per https://github.com/dlang/dmd/blob/master/src/posix.mak#L160. The "total" solution to that is to canonicalize and place the controlling flags in the dir names, i.e. (after removing `ENABLE`) in alphabetical order: `COVERAGE`, `DEBUG`, `LTO`, `PGO_GENERATE`, `PGO_USE`, `PROFILE`, `PROFILING`, `RELEASE`, `UNITTEST`. Then the dir name of a generated bunch of binaries would be something like `generated/posix64/DEBUG_PROFLE_UNITTEST`. The "final release" version should sit in `generated/$(OS)$(MODEL/` so flags do not affect tool developers. This scheme may be overkill even if technically appropriate. I think @WalterBright and @MartinNowak should chime in here because they'll be the first and most affected.